app/carbonapi/http_handlers.go

```diff
@@ -350,6 +350,9 @@ func (app *App) renderHandler(w http.ResponseWriter, r *http.Request) {
 					bs := backend.Filter(app.backends, request.Targets)
 					metrics, err := backend.Renders(ctx, bs, request)
 
+					// time in queue is converted to ms
+					app.prometheusMetrics.TimeInQueue.Observe(float64(request.Trace.Report()[2]) / 1000)
```

Comment:
It would be great to make return value from Report method as some struct (not array) and make some getters for structure fields. It would be better to make source more readable.